lib/matplotlib/type1font.py

```diff
@@ -35,9 +34,213 @@
 
 _log = logging.getLogger(__name__)
 
-# token types
-_TokenType = enum.Enum('_TokenType',
-                       'whitespace name string delimiter number')
+
+class _Token:
+    """
+    A token in a PostScript stream
+
+    Attributes
+    ----------
+    pos : int
+        position, i.e. offset from the beginning of the data
+
+    value : bytes
```

Comment:
I wonder if it may be simpler to just ascii-replace decode everything upfront and work on strs throughout?  This would e.g. remove the need to workaround bytes indexing semantics and let us write `if `text[pos] == "<"` instead of `text[pos:pos+1] == b"<"`, as well as the calls to `convert` midway through the parsing.